src/main/java/digital/slovensko/autogram/core/Autogram.java

```diff
@@ -56,22 +58,45 @@ public void checkPDFACompliance(SigningJob job) {
             return;
 
         ui.onWorkThreadDo(() -> {
-            var result = new PDFAStructureValidator().validate(job.getDocument());
+            // PDF/A doesn't support encryption
+            if (job.getDocument().hasOpenDocumentPassword()) {
+                ui.onUIThreadDo(() -> ui.onPDFAComplianceCheckFailed(job));
+                return;
+            }
+
+            var result = new PDFAStructureValidator().validate(job.getDocument().getDSSDocument());
             if (!result.isCompliant()) {
                 ui.onUIThreadDo(() -> ui.onPDFAComplianceCheckFailed(job));
             }
         });
     }
 
+    public void handleProtectedPdfDocument(AutogramDocument document) {
+        var protection = PDFUtils.determinePDFProtection(document.getDSSDocument());
+        if (protection == PDFUtils.PDFProtection.NONE)
+            return;
+
+        var password = ui.getDocumentPassword(document.getDSSDocument());
+        switch (protection) {
+            case OPEN_DOCUMENT_PASSWORD -> document.setOpenDocumentPassword(password);
+            case MASTER_PASSWORD -> document.setMasterPassword(password);
+        }
+    }
+
+    public SigningJob buildSigningJobFromFile(File file, Responder responder, boolean checkPDFACompliance, SignatureLevel signatureType, boolean isEn319132, TSPSource tspSource, boolean plainXmlEnabled) {
+        var document = SigningJob.createDSSFileDocumentFromFile(file);
+        handleProtectedPdfDocument(document);
+
+        var parameters = SigningJob.getParametersForFile(document, checkPDFACompliance, signatureType, isEn319132, tspSource, plainXmlEnabled);
+        return SigningJob.build(document, parameters, responder);
+    }
+
+    public void wrapInWorkThread(Runnable callback) {
+        ui.onWorkThreadDo(callback);
```

Comment:
Toto sa pouziva presne na 1 mieste, cize dajme inline, nijako to nezvysuje prehladnost, skor naopak.

Reply:
`ui` je private, preto som to spravil do autogram triedy priamo.